File: 4_Baseball_game/Golang/Main.go
package main

import (
	"fmt"
	"strconv"
)

func baseballGame(ops []string) int {
	var record = []int{}
	var total = 0

	for i := 0; i < len(ops); i++ {
		if _, err := strconv.Atoi(ops[i]); err == nil {
			num, _ := strconv.Atoi(ops[i])
			record = append(record, num)
		} else if ops[i] == "C" {
			record = record[0 : len(record)-1]
		} else if ops[i] == "D" {
			record = append(record, record[len(record)-1]*2)
		} else if ops[i] == "+" {
			record = append(record, record[len(record)-1]+record[len(record)-2])
		}
	}

	if len(record) > 0 {
		for i := 0; i < len(record); i++ {
			total += record[i]
		}
	}

	return total
}

func main() {

	x := baseballGame([]string{"1", "C"})
	fmt.Println(x)
}

File: 4_Baseball_game/JavaScript/Main.js
var baseballGame = function (ops) {
  let record = []
  let total = 0

  for (let i = 0; i < ops.length; i++) {
    if (!isNaN(Number(ops[i]))) {
      record.push(Number(ops[i]))
    } else if (ops[i] === "C") {
      record.pop()
    } else if (ops[i] === "D") {
      record.push(record[record.length - 1] * 2)
    } else if (ops[i] === "+") {
      record.push(record[record.length - 1] + record[record.length - 2])
    }
  }

  if (record.length) {
    for (let i = 0; i < record.length; i++) {
      total += record[i]
    }
  }

  return total
};

console.log(baseballGame(["1","C"]));